src/commonMain/kotlin/exchange.dydx.abacus/state/app/adaptors/V4TransactionErrors.kt

```diff
@@ -16,9 +16,45 @@ class V4TransactionErrors {
                 } else {
                     null
                 }
+            } else if (message?.startsWith("Query failed") == true) {
+                parseQueryResultErrorFromMessage(message)
             } else {
                 ParsingError(ParsingErrorType.BackendError, message ?: "Unknown error", null)
             }
         }
+
+        private fun parseQueryResultErrorFromMessage(message: String): ParsingError {
+            // Workaround: Regex match different query results until protocol can return codespace/code
+            parseSubaccountUpdateError(message)?.let { return it }
+            return ParsingError(ParsingErrorType.BackendError, "Unknown query result error", null)
+        }
+
+        private fun parseSubaccountUpdateError(message: String): ParsingError? {
+            val failedSubaccountUpdateResultPattern = Regex("""Subaccount with id \{[^}]+\} failed with UpdateResult:\s*([A-Za-z]+):""")
```

Comment:
That's quite hacky..   Let's hope Protocol wont change the error message any time soon..